processor/schemaprocessor/internal/translation/revision.go

```diff
@@ -0,0 +1,190 @@
+	Spans() Modifier
+
+	// SpanEvents update all incoming span event signals
+	// that are part of this revision
+	SpanEvents() Modifier
+
+	// Metrics update all incoming metric signals
+	// that are part of this reivision
+	Metrics() Modifier
+}
+
+type revision struct {
+	ver      *Version
+	all      Modifier
+	resource Modifier
+	logs     Modifier
+	spans    Modifier
+	events   Modifier
+	metrics  Modifier
+}
+
+var (
+	_ Revision = (*revision)(nil)
+)
+
+// newRevision processes the VersionDef and assigns the version to this revision
+// to allow sorting within a slice.
+// Since VersionDef uses custom types for various definitions, it isn't possible
+// to cast those values into the primitives so each has to be processed together.
+// Generics would be handy here.
+func newRevision(ver *Version, def ast.VersionDef) Revision {
+	r := &revision{
+		ver:      ver,
+		all:      noModify{},
+		resource: noModify{},
+		logs:     noModify{},
+		spans:    noModify{},
+		events:   noModify{},
+		metrics:  noModify{},
+	}
+
+	if len(def.All.Changes) > 0 {
+		mod := newModify()
+		for _, ch := range def.All.Changes {
+			if ch.RenameAttributes == nil {
+				continue
+			}
+			for k, v := range *ch.RenameAttributes {
+				mod.attrs[k] = v
+			}
+		}
+		r.all = mod
+	}
+
+	if len(def.Resources.Changes) > 0 {
+		mod := newModify()
+		for _, ch := range def.Resources.Changes {
+			if ch.RenameAttributes == nil {
+				continue
+			}
+			for k, v := range *ch.RenameAttributes {
+				mod.attrs[k] = v
+			}
+		}
+		r.resource = mod
+	}
+
+	if l := len(def.Logs.Changes); l > 0 {
+		mod := newModify()
+		for _, ch := range def.Logs.Changes {
+			if ch.RenameAttributes == nil {
+				continue
+			}
+			for k, v := range ch.RenameAttributes.AttributeMap {
+				mod.attrs[k] = v
+			}
+		}
+		r.logs = mod
+	}
+
+	if l := len(def.Metrics.Changes); l > 0 {
+		mods := make(modifications, l)
+		for i, ch := range def.Metrics.Changes {
+			mod := newModify()
+			if ch.RenameAttributes != nil {
+				for _, name := range ch.RenameAttributes.ApplyToMetrics {
+					mod.appliesTo[string(name)] = struct{}{}
+				}
+				for k, v := range ch.RenameAttributes.AttributeMap {
+					mod.attrs[k] = v
+				}
+			}
+			for k, v := range ch.RenameMetrics {
+				mod.names[string(k)] = string(v)
+			}
+			mods[i] = mod
+		}
+		r.metrics = mods
+	}
+	if l := len(def.Spans.Changes); l > 0 {
+		mods := make(modifications, l)
+		for i, ch := range def.Spans.Changes {
+			mod := newModify()
+			if ch.RenameAttributes != nil {
+				for _, name := range ch.RenameAttributes.ApplyToSpans {
+					mod.appliesTo[string(name)] = struct{}{}
+				}
+				for k, v := range ch.RenameAttributes.AttributeMap {
+					mod.attrs[k] = v
+				}
+			}
+			mods[i] = mod
+		}
+		r.spans = mods
+	}
+
+	if l := len(def.SpanEvents.Changes); l > 0 {
+		mods := make(modifications, l)
+		for i, ch := range def.SpanEvents.Changes {
+			mod := newModify()
+			if ch.RenameAttributes != nil {
```

Comment:
I think I now understand better the "actions":

So there is a forward change (from v1 -> v2) and a backwards change (v2 ->v1). Not sure how to name the Modifier interface but clearly needs some naming improvements there.

Reply:
Yeah, I am still scratching my head on what would be a nice name for the Modifier.

Perhaps something like:
- `SignalValueTranslator` 
- `ValueModifer`
- `RevisionUpdater`